### PR TITLE
Python package restructuring

### PR DESCRIPTION
**Pull request by g-bauer**
- Changed project layout (a bit) to match "uv" setups.
- added numpy as dependency (automatically installed when pulled from PyPi)
- added stub file for editor support outside of notebooks
- use MkDocs for documentation 

Todo:
- setup CI for wheels and docs (also, doctests)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,8 +3,12 @@ Cargo.lock
...
 *.ipynb
 __pycache__
-test_readme.py
\ No newline at end of file
+test_readme.py
+*.lock
+
+# venv
+/venv
+.venv
```
Comment (prehner) [inline .gitignore:14]: add the directory created my mkdoc here?

File: si-units/docs/index.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+sqms = ms**2
+print(sqms)
+```
+
+```
+[4, 9, 16] m²
+```
+
+When you divide a quantity by its unit, the value
+(i.e. `float` for scalars or `numpy.ndarray[flaot]` for arrays) is returned.
```
Comment (prehner) [inline si-units/docs/index.md:37]:
```suggestion
(i.e. `float` for scalars or `numpy.ndarray[float]` for arrays) is returned.
```

File: si-units/src/si_units/_core.pyi [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        Args:
+            other: The quantity to compare to.
+
+        Returns:
+            Wheter the units of the compared quantities are the same or not.
+        """
+        ...
+
+class SIArray1:
+    """Builder for SIObjects with numpy.narray as value storage."""
```
Comment (prehner) [inline si-units/src/si_units/_core.pyi:79]:
```suggestion
    """Builder for SIObjects with numpy.ndarray as value storage."""
```

**Review by prehner — CHANGES_REQUESTED** [review]
Very nice! The documentation looks crisp and having docstrings and no warnings for things that come out of * imports is a mahor improvement. Some things I can't really follow.

File: Cargo.toml [hunk 4]
```diff
@@ -1,3 +1,18 @@
...
+
+[workspace.package]
+version = "0.10.0"
+authors = [
+    "Philipp Rehner <prehner@ethz.ch>",
+    "Gernot Bauer <bauer@itt.uni-stuttgart.de>",
+]
+edition = "2021"
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/itt-ustutt/quantity"
```
[the remark this answers is not in the document]
Comment (g-bauer) [reply Cargo.toml:13]: Agreed. Had a different structure in mind when I changed this, but it is not needed and I removed it.

File: si-units/src/si_units/_core.pyi (hunk 3 above)
Comment (g-bauer) [reply si-units/src/si_units/_core.pyi]: Removed SIArray1 so this is no longer an issue

File: .gitignore (hunk 1 above)
Comment (g-bauer) [reply .gitignore:16]: Good idea! I added the `site` directory (amongst others that are built during packaging).

**Review by prehner — APPROVED** [review]

**Pull request merged**